staging/src/k8s.io/client-go/rest/config_test.go

```diff
@@ -100,6 +101,38 @@ func TestSetKubernetesDefaultsUserAgent(t *testing.T) {
 	}
 }
 
+func TestAdjustVersion(t *testing.T) {
+	assert := assert.New(t)
+	assert.Equal("1.2.3", adjustVersion("1.2.3-alpha4"))
+	assert.Equal("1.2.3", adjustVersion("1.2.3-alpha"))
+	assert.Equal("1.2.3", adjustVersion("1.2.3"))
+}
+
+func TestAdjustCommit(t *testing.T) {
+	assert := assert.New(t)
+	assert.Equal("1234567", adjustCommit("1234567890"))
+	assert.Equal("123456", adjustCommit("123456"))
+	assert.Equal("unknown", adjustCommit(""))
+}
+
+// This test expects to run on linux, and would fail with a Windows
+// path like C:\Users\Bob\Downloads\beans since its impl is os dependent.
+func TestAdjustCommand(t *testing.T) {
+	assert := assert.New(t)
+	assert.Equal("beans", adjustCommand("/home/bob/Downloads/junk/beans"))
```

Comment:
This test would be broken if run on windows. You need to use filepath.Join to construct a file path.